app/models/miq_expression.rb

```diff
@@ -371,56 +371,56 @@ def to_human
     self.class._to_human(@exp)
   end
 
-  def self._to_human(exp, options = {})
-    return exp unless exp.kind_of?(Hash) || exp.kind_of?(Array)
+  def self._to_human(expr, options = {})
```

Comment:
right, there is no shadowing in class methods. However, after changing scope of self._to_human() to instance scope, there would be. Also,  I think it would provide better readability  if parameters named consistently.